packages/app/src/cli/services/dev/app-events/app-event-watcher.ts

```diff
@@ -0,0 +1,279 @@
+    const newConfig = ext.configuration
+    if (oldConfig === undefined) return false
+    return JSON.stringify(oldConfig) !== JSON.stringify(newConfig)
+  })
+  const createdEvents = createdExtensions.map((ext) => ({type: EventType.Created, extension: ext}))
+  const deletedEvents = deletedExtensions.map((ext) => ({type: EventType.Deleted, extension: ext}))
+  const updatedEvents = updatedExtensions.map((ext) => ({type: EventType.UpdatedSourceFile, extension: ext}))
+  return {
+    app: newApp,
+    extensionEvents: [...createdEvents, ...deletedEvents, ...updatedEvents],
+    startTime: event.startTime,
+    path: event.path,
+  }
+}
+
+/**
+ * When the app.toml is updated:
+ * Reload the app and return the new app and the updated extensions in the event.
+ * Compare the old and new extensions (defined in the app tomle) to find the created, deleted and updated extensions.
+ */
+async function AppConfigUpdatedHandler({event, app, options}: HandlerInput): Promise<AppEvent> {
+  const newApp = await reloadApp(app, options)
+  const oldExtensions = app.realExtensions.filter((ext) => ext.configurationPath === app.configuration.path)
+  const oldExtensionsHandles = oldExtensions.map((ext) => ext.handle)
+  const newExtensions = newApp.realExtensions.filter((ext) => ext.configurationPath === app.configuration.path)
+  const newExtensionsHandles = newExtensions.map((ext) => ext.handle)
+
+  const createdExtensions = newExtensions.filter((ext) => !oldExtensionsHandles.includes(ext.handle))
+  const deletedExtensions = oldExtensions.filter((ext) => !newExtensionsHandles.includes(ext.handle))
+  const updatedExtensions = newExtensions.filter((ext) => {
+    const oldConfig = oldExtensions.find((oldExt) => oldExt.handle === ext.handle)?.configuration
+    const newConfig = ext.configuration
+    if (oldConfig === undefined) return false
+    return JSON.stringify(oldConfig) !== JSON.stringify(newConfig)
+  })
+  const createEvents = createdExtensions.map((ext) => ({type: EventType.Created, extension: ext}))
+  const deleteEvents = deletedExtensions.map((ext) => ({type: EventType.Deleted, extension: ext}))
+  const updateEvents = updatedExtensions.map((ext) => ({type: EventType.Updated, extension: ext}))
+  return {
+    app: newApp,
+    extensionEvents: [...createEvents, ...deleteEvents, ...updateEvents],
+    startTime: event.startTime,
+    path: event.path,
+  }
+}
+
+/**
+ * When the app.toml is deleted:
+ * Throw an error to exit the process.
+ */
+async function AppConfigDeletedHandler(_input: HandlerInput): Promise<AppEvent> {
+  // The user deleted the active app.toml, why would they do that? :(
+  throw new AbortError('The active app.toml was deleted, exiting')
+}
+
+/*
+ * Reload the app and returns it
+ * Prints the time to reload the app to stdout
+ */
+async function reloadApp(app: AppInterface, options: OutputContextOptions): Promise<AppInterface> {
```

Comment:
We're expecting `loadApp` to work without any input in this case, right -- Or would that be okay if it put up a prompt?

Reply:
During `dev` we can't show prompts, so this loadApp needs to work automatically. 

afaik, the loader doesn't have any prompts anyway right?

Reply:
It might? Maybe with the last-used logic? If it doesn't: great